File: src/main/java/com/goudourasv/domain/instructors/InstructorsService.java
package com.goudourasv.domain.instructors;

import com.goudourasv.data.instructors.InstructorsRepository;
import com.goudourasv.http.instructors.dto.InstructorCreate;
import com.goudourasv.http.instructors.dto.InstructorUpdate;

import javax.enterprise.context.ApplicationScoped;
import javax.transaction.Transactional;
import java.util.List;
import java.util.UUID;

@ApplicationScoped
public class InstructorsService {
    private InstructorsRepository instructorsRepository;

    public InstructorsService(InstructorsRepository instructorsRepository) {
        this.instructorsRepository = instructorsRepository;
    }

    @Transactional
    public List<Instructor> getInstructors(UUID institutionId) {
        List<Instructor> instructors = instructorsRepository.getInstructors(institutionId);
        return instructors;
    }

    @Transactional
    public Instructor getSpecificInstructor(UUID id) {
        Instructor instructor = instructorsRepository.getSpecificInstructor(id);
        return instructor;

    }

    @Transactional
    public Instructor createNewInstructor(InstructorCreate instructorCreate) {
        Instructor instructor = instructorsRepository.createNewInstructor(instructorCreate);
        return instructor;

    }

    @Transactional
    public boolean deleteSpecificInstructor(UUID id) {
        boolean deleted = instructorsRepository.deleteSpecificInstructor(id);
        return deleted;
    }

    @Transactional
    public Instructor replaceInstructor(UUID id, InstructorCreate instructorCreate) {
        Instructor instructor = instructorsRepository.replaceInstructor(id, instructorCreate);
        return instructor;

    }

    @Transactional
    public Instructor partiallyUpdateInstructor(InstructorUpdate instructorUpdate, UUID id) {
        Instructor instructor = instructorsRepository.partiallyUpdateInstructor(instructorUpdate, id);
        return instructor;
    }

}

File: src/main/java/com/goudourasv/domain/instructors/InstructorsService.kt
package com.goudourasv.domain.instructors

import com.goudourasv.data.instructors.InstructorsRepository
import com.goudourasv.http.instructors.dto.InstructorCreate
import com.goudourasv.http.instructors.dto.InstructorUpdate
import java.util.*
import javax.enterprise.context.ApplicationScoped
import javax.transaction.Transactional

@ApplicationScoped
class InstructorsService(private val instructorsRepository: InstructorsRepository) {
    @Transactional
    fun getInstructors(institutionId: UUID?): List<Instructor> {
        return instructorsRepository.getInstructors(institutionId)
    }

    @Transactional
    fun getSpecificInstructor(id: UUID): Instructor? {
        return instructorsRepository.getSpecificInstructor(id)
    }

    @Transactional
    fun createNewInstructor(instructorCreate: InstructorCreate): Instructor {
        return instructorsRepository.createNewInstructor(instructorCreate)
    }

    @Transactional
    fun deleteSpecificInstructor(id: UUID): Boolean {
        val deleted = instructorsRepository.deleteSpecificInstructor(id)
        return deleted
    }

    @Transactional
    fun replaceInstructor(id: UUID, instructorCreate: InstructorCreate): Instructor {
        return instructorsRepository.replaceInstructor(id, instructorCreate)
    }

    @Transactional
    fun partiallyUpdateInstructor(id: UUID, instructorUpdate: InstructorUpdate): Instructor {
        return instructorsRepository.partiallyUpdateInstructor(instructorUpdate, id)

    }


}